modelon/impact/client/client.py

```diff
@@ -1,20 +1,89 @@
 """This module provides an entry-point to the client APIs."""
 import logging
+from typing import List, Optional
+from dataclasses import dataclass
 from semantic_version import SimpleSpec, Version  # type: ignore
 import modelon.impact.client.configuration
-from modelon.impact.client.entities.workspace import WorkspaceDefinition, Workspace
-from modelon.impact.client.entities.project import ProjectDefinition, Project
+from modelon.impact.client.entities.project import Project, ProjectDefinition, VcsUri
+from modelon.impact.client.entities.workspace import (
+    WorkspaceDefinition,
+    Workspace,
+    VcsReference,
+)
 import modelon.impact.client.sal.service
 import modelon.impact.client.sal.exceptions
 import modelon.impact.client.credential_manager
 import modelon.impact.client.jupyterhub
+from modelon.impact.client.operations.workspace import WorkspaceImportOperation
 from modelon.impact.client.sal.uri import URI
 from modelon.impact.client import exceptions
 
 
 logger = logging.getLogger(__name__)
 
 
+@dataclass
+class Selection:
+    entry_id: str
+    project: Project
+
+    def to_dict(self):
+        return {'id': self.entry_id, 'project': {'id': self.project.id}}
+
+
+@dataclass
+class ProjectMatching:
+    entry_id: str
+    vcs_uri: str
+    projects: List[Project]
+
+    def get_selection(self, index: int) -> Selection:
+        return Selection(entry_id=self.entry_id, project=self.projects[index])
+
+    def make_selection_interactive(self) -> Selection:
+        if len(self.projects) == 1:
+            print(f"Only one project matches the URI {self.vcs_uri}!")
+            selection = self.get_selection(index=0)
+            print(
+                f"Resolving conflict automatically using project with ID: "
+                f"{selection.project.id} for repository with URI: {self.vcs_uri}."
+            )
+            return selection
+
+        print(f"Multiple existing projects match the URI {self.vcs_uri}!\n")
```

Comment:
I think it would be nicer to not have the line break after this statement, because you want it grouped with the choices that are printed after. The line break should be before this statement so that you make some space against what was printed before? So something like:
print(f"\nMultiple existing projects match the URI {self.vcs_uri}:")

Reply:
A suggestion to improve the statement so it is more clear for the user what is going on:
`The URI {self.vcs_uri} in the workspace definition matched with multiple projects during import:`